pkg/controller/certificates/certificate_controller.go

```diff
@@ -110,6 +112,12 @@ func NewCertificateController(
 	return cc
 }
 
+// WithDynamicCertReloader adds a dynamicCertReloader during initialization
+func (cc *CertificateController) WithDynamicCertReloader(dynamicCertReloader dynamiccertificates.ControllerRunner) *CertificateController {
```

Comment:
would it make sense to unit test `NewCertificateController(...).WithDynamicCertReloader(...)` in `certificate_controller_test.go` with a fake `ControllerRunner`?

Reply:
> would it make sense to unit test `NewCertificateController(...).WithDynamicCertReloader(...)` in `certificate_controller_test.go` with a fake `ControllerRunner`?

To see if the Run method is called?  Right now, the only unit test calling newcertcontroller doesn't even attempt a real Run.